SCAutolib/models/user.py

```diff
@@ -0,0 +1,153 @@
+from pathlib import Path
+from python_freeipa.client_meta import ClientMeta
+
+from SCAutolib import run, logger
+
+
+class User:
+    """
+    Generic class to represent system users.
+    """
+
+    def __init__(self, username: str, password: str, pin: str, card = None,
+                 cnf: Path = None, key: Path = None, cert: Path = None):
+
+        """
+        :param username: Username for the system user
+        :type: str
+        :param password: Password for the system user
+        :type str
+        :param pin: Smart card pin for the system user
+        :type str
+        :param card: Card to be associated with the user
+        :type Card
+        :param cnf: CNF file to be associated with the user
+        :type Path
+        :param key: Key to be assoicated with the user
+        :type Path
+        :param cert: Certificate to be associated with the user.
+        :type Path
+        """
+
+        self.username = username
+        self.password = password
+        self.pin = pin
+        self._card = card
+        self._cnf = cnf
+        self._key = key
+        self._cert = cert
+
+    @property
+    def card(self):
+        return self._card
+
+    @card.setter
+    def card(self, card: Path):
+        if self._card:
+            logger.error("Delete the existing card before adding a new one.")
+            raise ValueError("A card is already assigned to this user")
+        self._card = card
+
+    @card.deleter
+    def card(self):
+        logger.info("Deleting the existing card from {self.username}")
+        self._card = None
```

Comment:
Do we really need `deleter` for properties? What are the use cases for this decorator?